libvips/foreign/rawsave.c

```diff
@@ -116,49 +117,149 @@ vips_foreign_save_raw_build(VipsObject *object)
 	if (VIPS_OBJECT_CLASS(vips_foreign_save_raw_parent_class)->build(object))
 		return -1;
 
-	if ((raw->fd = vips__open_image_write(raw->filename, FALSE)) < 0 ||
-		vips_image_pio_input(save->in) ||
-		vips_sink_disc(save->in, vips_foreign_save_raw_write, raw))
+	if (vips_image_pio_input(save->in) ||
+		vips_sink_disc(save->in, vips_foreign_save_raw_block, raw))
 		return -1;
 
```

Comment:
You'll need to `_end()` the target after write, eg, add:

```C
    if (vips_target_end(raw->target))
        return -1;
```

Reply:
I think this will fix the testsuite failures as well.

Reply:
Corrected now, thanks!